pkg/encoding/v1/encoding.go

```diff
@@ -233,16 +302,37 @@ func (s *Service) UnmarshalYAML(unmarshal func(interface{}) error) error {
 
 type VolumeSize string
 
-// FIXME: add VolumeSize parsing/validation
+func (vs *VolumeSize) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
`UnmarshalYAML` for `ResourceName` does validation on top of unmarshalling . 
But in this case you are not doing anything special, you are just calling `unmarshal` function on `string` type. You don't need custom  `UnmarshalYAML` function in this case